### DISCUSSION TITLE
Can we use different classifier in Node2Vec link prediction

### DISCUSSION THREAD

**Discussion by shahfasal** [stellargraph/stellargraph#1903] [Q&A]
Hi, I am using Stellargraph for link prediction on a knowledge graph using Node2Vec embeddings as shown in (https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/node2vec-link-prediction.html). I see in the code Logistic regression is used. My question is simple can I change the classifier in the code to some other algorithm? Let's say, Naive Bayes!!! I hope there will be no complications and this should be doable.
Thanks

**Comment by adocherty** [comment]
Hi,

Thanks for your question.  You can definitely use other classifiers  other than logistic regression.  Using node2vec for node classification is a two step process of learning a node embedding (a feature vector) for each node, then applying an arbitrary classifier to the embedding vectors to learn the node labels.  Most of the classifiers in Scikit-Learn should work just by substituting the alternative classifier in the code. 

I hope this helps!
